src/Any/Signer.cpp

```diff
@@ -4,167 +4,225 @@
-            if (output.success()) {
-                message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
-                auto signerOutput = Nebulas::Signer(load(message.chain_id())).sign(message);
-                auto signature = signerOutput.signature();
-                output.set_output(hex(signature.begin(), signature.end()));
-            }
-            break;
+        break;
+    }
+    case TWCoinTypeNebulas: {
+        Nebulas::Proto::SigningInput message;
+        parse(transaction, &message, output);
+        if (output.success()) {
+            message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+            auto signerOutput = Nebulas::Signer(load(message.chain_id())).sign(message);
+            auto signature = signerOutput.signature();
+            output.set_output(hex(signature.begin(), signature.end()));
         }
-        case TWCoinTypeTron: {
-            Tron::Proto::SigningInput message;
-            parse(transaction, &message, output);
-            if (output.success()) {
-                message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
-                auto signerOutput = Tron::Signer::sign(message);
-                output.set_output(signerOutput.json());
-            }
-            break;
+        break;
+    }
+    case TWCoinTypeTron: {
+        Tron::Proto::SigningInput message;
+        parse(transaction, &message, output);
+        if (output.success()) {
+            message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+            auto signerOutput = Tron::Signer::sign(message);
+            output.set_output(signerOutput.json());
+        }
+        break;
+    }
+    case TWCoinTypeVeChain: {
+        VeChain::Proto::SigningInput message;
+        parse(transaction, &message, output);
+        if (output.success()) {
+            message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+            auto signerOutput = VeChain::Signer::sign(message);
+            auto encoded = signerOutput.encoded();
+            output.set_output(hex(encoded.begin(), encoded.end()));
         }
-        case TWCoinTypeVeChain: {
-            VeChain::Proto::SigningInput message;
-            parse(transaction, &message, output);
-            if (output.success()) {
-                message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
-                auto signerOutput = VeChain::Signer::sign(message);
-                auto encoded = signerOutput.encoded();
-                output.set_output(hex(encoded.begin(), encoded.end()));
+        break;
+    }
+    case TWCoinTypeHarmony: {
+        Harmony::Proto::SigningInput message;
+        parse(transaction, &message, output);
+        if (output.success()) {
+            message.set_private_key(privateKey.bytes.data(), privateKey.bytes.size());
+            Harmony::Proto::SigningOutput signerOutput;
```

Comment:
move this code inside C++ signer, so AnySigner and TWHarmonySigner can benefit

Reply:
This code should be moved for sure